src/main/java/de/unistuttgart/overworldbackend/data/CourseDTO.java

```diff
@@ -21,13 +22,24 @@ public class CourseDTO {
   @NotNull
   String courseName;
 
-  String description;
+  @Pattern(regexp = "^(WS|SS)-[0-9][0-9]+$")
```

Comment:
LGTM once you extract this String literal into a constant to use always.

Reply:
where should this constant be stored?

Reply:
That is a better question...
I think `Course` is the most fitting, because then the line will always be `@Pattern(regexp = Course.SEMESTER_PATTERN)`